src/compiler/checker.ts

```diff
@@ -11656,7 +11656,7 @@ namespace ts {
                 if (!node) return false;
                 switch (node.kind) {
                     case SyntaxKind.Identifier:
-                        return (<Identifier>node).escapedText === "arguments" && isExpressionNode(node);
+                        return (<Identifier>node).escapedText === argumentsSymbol.escapedName && getResolvedSymbol(<Identifier>node) === argumentsSymbol;

```

Comment:
I think it would be simpler to check that the parent node type isn't PropertyAccessExpression / ElementAccessExpression? Does that work?

Reply:
Thanks for the review. Good point. Just to be sure, do we need to add `...args` for the this case: [argumentsReferenceInConstructor7_Js](https://github.com/microsoft/TypeScript/blob/75ed47c259884a48646b3a32139d20679f20c9b0/tests/baselines/reference/argumentsReferenceInConstructor7_Js.js) and omit `...args` for this case [argumentsReferenceInConstructor4_Js](https://github.com/microsoft/TypeScript/blob/75ed47c259884a48646b3a32139d20679f20c9b0/tests/baselines/reference/argumentsReferenceInConstructor4_Js.js)?